src/types/redis_set.cc

```diff
@@ -338,19 +338,43 @@ rocksdb::Status Set::Inter(const std::vector<Slice> &keys, std::vector<std::stri
   for (const auto &member : target_members) {
     member_counters[member] = 1;
   }
+
+  bool has_limit = limit != 0;
+  bool limited = false;
   for (size_t i = 1; i < keys.size(); i++) {
     s = Members(keys[i], &target_members);
     if (!s.ok() || target_members.empty()) return s;
     for (const auto &member : target_members) {
-      if (member_counters.find(member) == member_counters.end()) continue;
+      if (member_counters.count(member) == 0) continue;
       member_counters[member]++;
+      if (has_limit && member_counters[member] == keys.size()) {
+        members->emplace_back(member);
+        if (--limit == 0) {
+          limited = true;
+          break;
+        }
+      }
     }
+
+    if (limited) break;
   }
-  for (const auto &iter : member_counters) {
-    if (iter.second == keys.size()) {  // all the sets contain this member
-      members->emplace_back(iter.first);
+
+  if (!has_limit) {
+    for (const auto &iter : member_counters) {
+      if (iter.second == keys.size()) {  // all the sets contain this member
+        members->emplace_back(iter.first);
+      }
     }
   }
+
+  return rocksdb::Status::OK();
+}
+
+rocksdb::Status Set::InterCard(const std::vector<Slice> &keys, uint64_t limit, int *ret) {
```

Comment:
Maybe not related to this PR, but there are two points:
- `ret` is not a descriptive word
- `int` type is not fit here: the return variable `members.size()` is typed `size_t`

Reply:
We find a few functions with `int *ret` in [redis_set.cc](https://github.com/apache/incubator-kvrocks/blob/unstable/src/types/redis_set.cc). However the result of `ret` is unsinged long(size_t)

Reply:
@PragmaTwice we exec the statement `*output = redis::Integer(ret);` to return the result to the client in cmd_set.cc.
https://github.com/apache/incubator-kvrocks/blob/fb0e3d42031a6e0691c7a736711d0a01d72bdb1a/src/server/redis_reply.h#L38-L41

Add the `Integer` func checks whether compatibility with `int` is required(Of course, `size_t` is right). I see src/types/*.h, many funcs with this situation use `int`.